docs/testing/runtimes.md

```diff
@@ -0,0 +1,117 @@
+
+All of these runtimes include an HTML rendering engine and provide support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [Full CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage), and cookies. 
+
+A browser runtime lifespan varies depending on the feature that it implements.
+
+- When an add-in with a task pane is launched a browser runtime starts. It shuts down when the add-in is closed.
+- When a dialog is opened, a browser runtime starts. It shuts down when the dialog is closed.
+- When a function command is executed (which happens when a user selects its button or menu item), a browser runtime starts, unless it is a shared runtime that is already running. If it is a shared runtime it shuts down when the add-in is closed. If it is an unshared runtime, it shuts down when the first of the following occurs:
+ 
+  - The function command calls the `completed` method of its event parameter.
+  - 5 minutes has elapsed since the triggering event. (If a dialog was opened in the function command and it is still open when the parent runtime times-out, the dialog runtime stays running until the dialog is closed.)
+
+- When an Excel custom function is using a shared runtime, then a browser type runtime starts when the custom function calculates if the shared runtime has not already started for some other reason. It shuts down when the add-in is closed.
+
+> [!NOTE]
+> When a runtime is being [shared](#shared-runtime), it is possible to close the task pane without shutting down the add-in. See [Show or hide the task pane of your Office Add-in](../develop/show-hide-add-in.md) for more information.
+
+### Shared runtime
+
+A "shared runtime" isn't a type of runtime. It refers to a [browser type runtime](#browser-runtime) that is being shared by features of the add-in that would otherwise each have their own runtime. Specifically, you have the option of configuring the add-in's task pane and function commands to share a runtime. In an Excel add-in, you also can configure a custom function to share the runtime of a task pane or function command or both. When you do this, the custom function is running in a browser type runtime, instead of a [JavaScript-only runtime](#javascript-only-runtime) as it otherwise would. See [Configure your add-in to use a shared runtime](../develop/configure-your-add-in-to-use-a-shared-runtime.md) for information about the benefits and limitations of sharing runtimes and instructions for configuring the add-in to use a shared runtime. 
```

Comment:
I think we should also specify the property of the SharedRuntime that the runtime will be running in background for the duration of the document even if the add-in is closed.

Reply:
done